applovin_max/android/src/main/java/com/applovin/applovin_max/AppLovinMAX.java

```diff
@@ -1253,18 +1299,58 @@ private void positionAdView(String adUnitId, MaxAdFormat adFormat)
             return;
         }
 
-        final String adViewPosition = mAdViewPositions.get( adUnitId );
         final RelativeLayout relativeLayout = (RelativeLayout) adView.getParent();
         if ( relativeLayout == null )
         {
             e( adFormat.getLabel() + "'s parent does not exist" );
             return;
         }
 
-        // Size the ad
+        final DisplayMetrics displayMetrics = new DisplayMetrics();
+        getCurrentActivity().getWindowManager().getDefaultDisplay().getMetrics( displayMetrics );
+
+        final String adViewPosition = mAdViewPositions.get( adUnitId );
         final AdViewSize adViewSize = getAdViewSize( adFormat );
-        final int width = AppLovinSdkUtils.dpToPx( getCurrentActivity(), adViewSize.widthDp );
-        final int height = AppLovinSdkUtils.dpToPx( getCurrentActivity(), adViewSize.heightDp );
+        final boolean isAdaptiveBannerDisabled = mDisabledAdaptiveBannerAdUnitIds.contains( adUnitId );
+        final boolean isWidthDpOverridden = mAdViewWidths.containsKey( adUnitId );
+
+        //
+        // Determine ad width
+        //
+        final int adViewWidthDp;
+
+        // Check if publisher has overridden width as dp
+        if ( isWidthDpOverridden )
+        {
+            adViewWidthDp = mAdViewWidths.get( adUnitId );
+        }
+        else if ( TOP_CENTER.equalsIgnoreCase( adViewPosition ) || BOTTOM_CENTER.equalsIgnoreCase( adViewPosition ) )
+        {
+            int adViewWidthPx = displayMetrics.widthPixels;
+            adViewWidthDp = AppLovinSdkUtils.pxToDp( getCurrentActivity(), adViewWidthPx );
+        }
+        else
+        {
+            adViewWidthDp = adViewSize.widthDp;
+        }
+
+        //
+        // Determine ad height
+        //
+        final int adViewHeightDp;
+
+        if ( ( adFormat == MaxAdFormat.BANNER || adFormat == MaxAdFormat.LEADER ) && !isAdaptiveBannerDisabled )
+        {
+            adViewHeightDp = adFormat.getAdaptiveSize( adViewWidthDp, getCurrentActivity() ).getHeight();
+        }
+        else
+        {
+            adViewHeightDp = adViewSize.heightDp;
+        }
+
+        // Size the ad
+        final int width = AppLovinSdkUtils.dpToPx( getCurrentActivity(), adViewWidthDp );
+        final int height = AppLovinSdkUtils.dpToPx( getCurrentActivity(), adViewHeightDp );
```

Comment:
Can we keep consistent with RN and name `widthPx` / `heightPx`? It makes it extremely clear the unit we are using as well.